plugins/modules/ome_network_vlan_info.py

```diff
@@ -0,0 +1,248 @@
+                    "Name": "Bronze"
+                },
+                "VendorCode": "GeneralPurpose"
+            },
+            "UpdatedBy": null,
+            "UpdatedTime": "2020-09-02 18:48:42.129",
+            "VlanMaximum": 111,
+            "VlanMinimum": 111
+        },
+        {
+            "CreatedBy": "admin",
+            "CreationTime": "2020-09-02 18:49:11.507",
+            "Description": "Description of Logical Network - 2",
+            "Id": 20058,
+            "InternalRefNWUUId": "e46ccb3f-ef57-4617-ac76-46c56594005c",
+            "Name": "Network VLAN - 2",
+            "Type": {
+                "Description": "This is the network for general purpose traffic. QOS Priority : Silver.",
+                "Id": 2,
+                "Name": "General Purpose (Silver)",
+                "NetworkTrafficType": "Ethernet",
+                "QosType": {
+                    "Id": 3,
+                    "Name": "Silver"
+                },
+                "VendorCode": "GeneralPurpose"
+            },
+            "UpdatedBy": null,
+            "UpdatedTime": "2020-09-02 18:49:11.507",
+            "VlanMaximum": 112,
+            "VlanMinimum": 112
+        }
+    ]
+}
+error_info:
+  description: Details of the HTTP Error.
+  returned: on HTTP error
+  type: dict
+  sample: {
+    "error": {
+      "code": "Base.1.0.GeneralError",
+      "message": "A general error has occurred. See ExtendedInfo for more information.",
+      "@Message.ExtendedInfo": [
+        {
+          "MessageId": "GEN1234",
+          "RelatedProperties": [],
+          "Message": "Unable to process the request because an error occurred.",
+          "MessageArgs": [],
+          "Severity": "Critical",
+          "Resolution": "Retry the operation. If the issue persists, contact your system administrator."
+        }
+      ]
+    }
+  }
+'''
+
+import json
+from ssl import SSLError
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.dellemc.openmanage.plugins.module_utils.ome import RestOME
+from ansible.module_utils.six.moves.urllib.error import URLError, HTTPError
+from ansible.module_utils.urls import ConnectionError, SSLValidationError
+
+# Base URI to fetch all logical networks information
+NETWORK_VLAN_BASE_URI = "NetworkConfigurationService/Networks"
+NETWORK_TYPE_BASE_URI = "NetworkConfigurationService/NetworkTypes"
+QOS_TYPE_BASE_URI = "NetworkConfigurationService/QosTypes"
+
+# Module Success Message
+MODULE_SUCCESS_MESSAGE = "Successfully retrieved the network VLAN information."
+
+# Module Failure Messages
+MODULE_FAILURE_MESSAGE = "Failed to retrieve the network VLAN information."
+NETWORK_VLAN_NAME_NOT_FOUND = "Provided network VLAN with name - '{0}' does not exist."
+
+
+def clean_data(data):
+    """
+    data: A dictionary.
+    return: A data dictionary after removing items that are not required for end user.
+    """
+    for k in ['@odata.id', '@odata.type', '@odata.context', '@odata.count']:

```

Comment:
'@data.count' should exist when pagination supported so that end-user can easily access the total number of available VLAN information

Reply:
Hi Sajna, I believe @odata.count would not require, user can just use a jinja "length" filter to get the count.

Further, This will even reduce false counts due to API inconsistencies wrt @odata.count, if any too.

Reply:
@deepakjoshishri @odata.count is to make sure the number of values returned is correct. We had an issue in ome_device_info where values returned wrongly whereas count returned correctly due to pagination issue. So it is better to return @odata.count as part of the output to confirm the values are matchup with the count.

Reply:
@Sajna-Shetty I confirm that, I have tested this on max available number of network VLAN(s) in OME i.e., 4074 as per my experimentation and the module seems to work fine, Further to ensure proper details are fetched when more than 100 network VLAN(s) are present, have added a $top with safe limit of 9999.

Thanks for pointing me out on a possible bug case.

Reply:
I have updated the same changes in another PR for devel branch #164 , Let me know if any more changes are expected.